packages/mermaid/src/mermaidAPI.ts

```diff
@@ -60,6 +60,15 @@ export interface ParseOptions {
   suppressErrors?: boolean;
 }
 
+export type ParseResult =
+  | {
+      /**
+       * The diagram type, e.g. 'flowchart', 'sequence', etc.
+       */
+      diagramType: string;
+    }
+  | false;
```

Comment:
I'd prefer for this type to be:

```ts
export type ParseResult = {
  /**
   * The diagram type, e.g. 'flowchart', 'sequence', etc.
   */
  diagramType: string;
};
```

And then we change the type for `parse()` to (may need to modify this slightly to get TypeScript's function overloading working):

```ts
async function parse(text: string, parseOptions?: {supressErrors: false}): Promise<ParseResult>;
async function parse(text: string, parseOptions?: ParseOptions): Promise<ParseResult | false> {
```

That way TypeScript will know that this function never returns `false` unless `{supressErrors: true}` is set.

It will also be much easier for users (and us) to declare a variable to be of type `ParseResult`!

Reply:
That's a wonderful approach.. Done!